crates/db_views/src/local_user_view.rs

```diff
@@ -13,132 +13,175 @@ use lemmy_db_schema::{
   traits::Crud,
   utils::{
     functions::{coalesce, lower},
-    DbConn,
+    get_conn,
     DbPool,
-    ListFn,
-    Queries,
-    ReadFn,
   },
 };
 use lemmy_utils::error::{LemmyError, LemmyErrorExt, LemmyErrorType, LemmyResult};
 use std::future::{ready, Ready};
 
-enum ReadBy<'a> {
-  Id(LocalUserId),
-  Person(PersonId),
-  Name(&'a str),
-  NameOrEmail(&'a str),
-  Email(&'a str),
-  OAuthID(OAuthProviderId, &'a str),
+#[diesel::dsl::auto_type]
+fn joins() -> _ {
+  local_user::table
+    .inner_join(local_user_vote_display_mode::table)
+    .inner_join(person::table)
+    .inner_join(person_aggregates::table.on(person::id.eq(person_aggregates::person_id)))
 }
 
-enum ListMode {
-  AdminsWithEmails,
-}
-
-fn queries<'a>(
-) -> Queries<impl ReadFn<'a, LocalUserView, ReadBy<'a>>, impl ListFn<'a, LocalUserView, ListMode>> {
-  let selection = (
-    local_user::all_columns,
-    local_user_vote_display_mode::all_columns,
-    person::all_columns,
-    person_aggregates::all_columns,
-  );
-
-  let read = move |mut conn: DbConn<'a>, search: ReadBy<'a>| async move {
-    let mut query = local_user::table.into_boxed();
```

Comment:
All of these moves, asyncs, boxes, ReadFn and ListFn, and Queries trait won't be necessary anymore.